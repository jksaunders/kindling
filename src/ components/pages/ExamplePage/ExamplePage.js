import React from 'react';
import styled from 'styled-components';
import { Layout, Typography } from '@jksaunders/bonfire';

const Content = styled(Layout)`
  background-color: rgba(255, 255, 255, 0.7);
`;

const ExamplePage = () => (
  <Content height="auto" width="100%" padding="16px" centered>
    <Typography.Body1>
      Example page!
    </Typography.Body1>
  </Content>
);

export default ExamplePage;