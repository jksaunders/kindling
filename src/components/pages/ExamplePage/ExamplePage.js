import React from 'react';
import styled from 'styled-components';
import { Layout, Typography } from '@jksaunders/bonfire';
import { observer } from 'mobx-react-lite';
import useStores from '@stores';

const Content = styled(Layout)`
  background-color: rgba(255, 255, 255, 0.7);
`;

const ExampleStoreUsage = observer(() => {
  const { authStore } = useStores();
  // eslint-disable-next-line no-console
  console.log(authStore.isLoggedIn);
  return null;
});

const ExamplePage = () => (
  <Content height="auto" width="100%" padding="16px" centered>
    <Typography>I&apos;m an example page!</Typography>
    <ExampleStoreUsage />
  </Content>
);

export default ExamplePage;
