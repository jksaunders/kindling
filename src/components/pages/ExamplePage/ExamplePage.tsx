import { observer } from 'mobx-react-lite';
import React, { FC } from 'react';
import styled from 'styled-components';
import useStores from '../../../stores';

const StyledComponentsExample = styled.div`
  border-radius: 10px;
  box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.1);
`;

const StoresExampleUsage = observer(() => {
  const { userStore } = useStores();
  return <div>{userStore.name}</div>;
});

const ExamplePage: FC = () => (
  <StyledComponentsExample>
    I&apos;m an example page!
    <StoresExampleUsage />
  </StyledComponentsExample>
);

export default ExamplePage;
