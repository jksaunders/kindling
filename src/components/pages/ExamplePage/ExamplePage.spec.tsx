import React from 'react';
import { render, screen } from '@testing-library/react';
import ExamplePage from './ExamplePage';

test('renders', async () => {
  const {
    container: { firstChild },
  } = render(<ExamplePage />);
  await screen.findByText("I'm an example page!");
  expect(screen.getByText("I'm an example page!")).toBeInTheDocument();
  expect(screen.getByText("I'm an example page!")).toHaveStyle(
    `border-radius: 10px`
  );
  expect(firstChild).toHaveStyle(`border-radius: 10px`);
  expect(firstChild).toMatchSnapshot();
});
