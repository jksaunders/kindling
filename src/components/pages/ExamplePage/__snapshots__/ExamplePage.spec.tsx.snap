// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  border-radius: 10px;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.1);
}

<div
  class="c0"
>
  I'm an example page!
</div>
`;
